### DISCUSSION TITLE
What's the best way to interact with a strategy while it's running to update its configuration/parameters and exit positions?

### DISCUSSION THREAD

**Discussion by eliahburns** [nautechsystems/nautilus_trader#1080] [Q&A] [answered]
I'm new to Nautilus and in the process of confirming it's viable for my use case. 

If I use it, I would like to build out a UI for my strategy where I can update its parameters as it runs and also work out of positions, or adjust stop losses and entry orders, manually some percentage of the time. 

What would be the best way to implement the communication of this logic? I've looked through the examples but haven't seen one yet that takes real-time user input to adjust profit targets, positions, or the general parameters that the strategy is running with.

**Comment by limx0** [comment, marked as answer]
Hey @eliahburns - there's a couple of choices here: 

1) In Nautilus this would typically be done through an adapter (similar to data and execution connections), but one that simple receives messages and passes them onto the strategy.
2) Strategies are pure python, so you can basically implement any sort of communication layer inside your strategy that you would like. 

These aren't however, built-in, because they're quite advanced and can take on many forms. Hope this helps.

**Reply by eliahburns** [reply]
Perfect, I thought an adapter might be the tool for the job to receive user updates to the strategy. 

Am I able to send updates from the strategy back through the adapter (to then forward to my UI)? Or is there a better pattern for this?

**Reply by limx0** [reply]
Depends what kind of stuff you want to send back - basically every event and piece of data is emitted on the message bus and captured to streaming feather files (if you set the `streaming=StreamingConfig()` on either backtest or live), which should cover 99% of use cases.

Users typically either tail these files or connect to the message bus and forward events/data elsewhere from there. An example might look like: 

```python
# Setup TradingNode / BacktestNode
node = TradingNode(config=config)

# Forward msgbus events/data
def publish(msg: Union[Event, Data]):
    # Do something with message

node.trader._msgbus.subscribe(topic="*", callback=publish)

```

**Reply by eliahburns** [reply]
Awesome, thanks, tailing the feather files sounds like the way to go 👍
